packages/web-sdk/src/instrumentations/webVitals/webVitalsWithAttribution.ts

```diff
@@ -0,0 +1,147 @@
+    onINP((metric) => {
+      const values = this.buildInitialValues(metric);
+      this.addIfPresent(values, 'interaction_time', metric.attribution.interactionTime);
+      this.addIfPresent(values, 'presentation_delay', metric.attribution.presentationDelay);
+      this.addIfPresent(values, 'input_delay', metric.attribution.inputDelay);
+      this.addIfPresent(values, 'processing_duration', metric.attribution.processingDuration);
+      this.addIfPresent(values, 'next_paint_time', metric.attribution.nextPaintTime);
+
+      const context = this.buildInitialContext(metric);
+      this.addIfPresent(context, loadState, metric.attribution.loadState);
+      this.addIfPresent(context, 'interaction_target', metric.attribution.interactionTarget);
+      this.addIfPresent(context, 'interaction_type', metric.attribution.interactionType);
+
+      this.pushMeasurement(values, context);
+    });
+  }
+
+  private measureLCP(): void {
+    onLCP((metric) => {
+      const values = this.buildInitialValues(metric);
+      this.addIfPresent(values, 'element_render_delay', metric.attribution.elementRenderDelay);
+      this.addIfPresent(values, 'resource_load_delay', metric.attribution.resourceLoadDelay);
+      this.addIfPresent(values, 'resource_load_duration', metric.attribution.resourceLoadDuration);
+      this.addIfPresent(values, timeToFirstByte, metric.attribution.timeToFirstByte);
+
+      const context = this.buildInitialContext(metric);
+      this.addIfPresent(context, 'element', metric.attribution.element);
+
+      this.pushMeasurement(values, context);
+    });
+  }
+
+  private measureTTFB(): void {
+    onTTFB((metric) => {
+      const values = this.buildInitialValues(metric);
+      this.addIfPresent(values, 'dns_duration', metric.attribution.dnsDuration);
+      this.addIfPresent(values, 'connection_duration', metric.attribution.connectionDuration);
+      this.addIfPresent(values, 'request_duration', metric.attribution.requestDuration);
+      this.addIfPresent(values, 'waiting_duration', metric.attribution.waitingDuration);
+      this.addIfPresent(values, 'cache_duration', metric.attribution.cacheDuration);
+
+      const context = this.buildInitialContext(metric);
+
+      this.pushMeasurement(values, context);
+    });
+  }
+
+  private buildInitialValues(metric: Metric): Values {
+    const indicator = metric.name.toLowerCase();
+    return {
+      [indicator]: metric.value,
+      delta: metric.delta,
+    };
+  }
+
+  private buildInitialContext(metric: Metric): Context {
+    const navigationEntryId = getItem(NAVIGATION_ID_STORAGE_KEY, webStorageType.session) ?? 'unknown';
+
+    return {
+      id: metric.id,
+      rating: metric.rating,
+      navigation_type: metric.navigationType,
+      navigation_entry_id: navigationEntryId,
+    };
+  }
+
+  private pushMeasurement(values: Values, context: Context): void {
+    const type = 'web-vitals';
+    this.corePushMeasurement({ type, values }, { context });
```

Comment:
Let's add the name of the web vital.


```suggestion
    this.corePushMeasurement({ name: <name of web vital> ,type, values }, { context });
```

Reply:
There is no such key `name` in the interface but I can pass this `name` via `context`, does this work for you?

Refs: https://github.com/grafana/faro-web-sdk/blob/cc21bd11dc3c2e85cff7bcc7ade70e464af75ccc/packages/core/src/api/measurements/types.ts#L7-L14

Reply:
Argh dang of course. 
Then let's keep it as it is. This leaves us compliant with the base web-vitals.

For references these are the resulting logs:

<img width="639" alt="image" src="https://github.com/grafana/faro-web-sdk/assets/47627413/66b4a1df-c54a-4aae-978d-a9a2db74fa8b">

<img width="650" alt="image" src="https://github.com/grafana/faro-web-sdk/assets/47627413/6b75ce89-be4d-4900-b650-ba0cd4da06d9">